File: 4_Dicas/4.5_Code_profiling/Ex1.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>

#define N 10000000

int main(int argc, char *argv[])
{
	int *x;
	int *y;
	int i;
	double dist_eucl, media_x, media_y;

	x = (int *)malloc(N*sizeof(int));
	y = (int *)malloc(N*sizeof(int));

	for(i=0; i<N; i++)
		x[i] = 100*((char)random())+5;

	for(i=0; i<N; i++)
		y[i] = 100*((char)random())+5;

	for(media_x=0.0, i=0; i<N; i++)
		media_x += ((double)x[i])/((double)N);

	for(media_y=0.0, i=0; i<N; i++)
		media_y += ((double)y[i])/((double)N);

	for(dist_eucl=0.0, i=0; i<N; i++)
		dist_eucl += (((double)x[i])-((double)y[i]))*
			(((double)x[i])-((double)y[i]));
	dist_eucl = sqrt(dist_eucl);


	printf("Resultado %s:\n", argv[0]);
	printf("   Media(x) = %3.1f\n",
		media_x);
	printf("   Media(y) = %3.1f\n",
		media_y);
	printf("   Dist_euclidiana(x,y) = %3.1f\n\n",
		dist_eucl);
	free(x);
	free(y);
	return 0;
}

File: 4_Dicas/4.6_Code_profiling/Octave/Ex1.m
N = 1000000;

x = [];
y = [];
for k = 0:(N-1)
	x(k+1) = floor(1000/bitand(k+1,(2^32)-1));
	y(k+1) = floor(1000/bitand(3*k*k+5*k+1,(2^32)-1));
end

media_x = 0;
for k = 1:N
	media_x = media_x + x(k)/N;
end

media_y = 0;
for k = 1:N
	media_y = media_y + y(k)/N;
end

dist_eucl = 0;
for k = 1:N
	dist_eucl = dist_eucl + (x(k)-y(k))^2;
end
dist_eucl = sqrt(dist_eucl);

disp("Resultado:")
disp(["   Media(x) = " num2str(media_x)]);
disp(["   Media(y) = " num2str(media_y)]);
disp(["   Dist_euclidiana(x,y) = " num2str(dist_eucl)]);